django/core/validators.py

```diff
@@ -83,9 +83,9 @@ class URLValidator(RegexValidator):
     ipv6_re = r'\[[0-9a-f:\.]+\]'  # (simple regex, validated later)
 
     # Host patterns
-    hostname_re = r'[a-z' + ul + r'0-9](?:[a-z' + ul + r'0-9-]*[a-z' + ul + r'0-9])?'
```

Comment:
I'd add a comment as done for EmailValidator: "# Max length for domain name labels is 63 characters per RFC 1034."